pkg/virt-handler/cache/cache.go

```diff
@@ -76,66 +75,68 @@ type ghostRecord struct {
 	UID        types.UID `json:"uid"`
 }
 
-var ghostRecordGlobalCache map[string]ghostRecord
-var ghostRecordGlobalMutex sync.Mutex
-var checkpointManager IterableCheckpointManager
-
-func InitializeGhostRecordCache(directoryPath string) error {
-	ghostRecordGlobalMutex.Lock()
-	defer ghostRecordGlobalMutex.Unlock()
+type GhostRecordStore struct {
+	cache             map[string]ghostRecord
+	mutex             sync.Mutex
```

Comment:
nit: just an idea:
```golang
type GhostRecordStore struct {
	cache map[string]ghostRecord
	checkpointManager checkpoint.CheckpointManager
	sync.Mutex
}
```
Avoiding navigate to `.mutex` every time.

Reply:
Done